crates/biome_js_analyze/src/lint/nursery/use_consistent_string_functions.rs

```diff
@@ -0,0 +1,141 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+};
+use biome_console::markup;
+use biome_js_factory::make;
+use biome_js_syntax::JsStaticMemberExpression;
+use biome_rowan::{BatchMutationExt, TextRange, TokenText};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Enforce the use of consistent string functions.
+    ///
+    /// This rule ensures that developers use consistent string functions by preferring:
+    ///
+    /// - `trimStart()` over `trimLeft()`,
+    /// - `trimEnd()` over `trimRight()`,
+    /// - `slice()` over `substr()` and `substring()`.
+    ///
+    /// While `trimLeft()` and `trimRight()` are aliases for `trimStart()` and `trimEnd()`, using the latter helps maintain consistency and uses direction-independent wording.
+    /// Similarly, `slice()` is preferred over `substr()` and `substring()` because it is a more popular option with clearer behavior, and it has a consistent counterpart in arrays.
```

Comment:
We should likely add link to MDN about differences between substring/slice https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring#the_difference_between_substring_and_substr